rust/nasl-c-lib/src/nasl_aes_mac_gcm.rs

```diff
@@ -0,0 +1,35 @@
+use libc::{c_int, c_uint, size_t};
+
+extern "C" {
+    fn nasl_aes_mac_gcm(
+        data: *const u8,
+        datalen: size_t,
+        key: *const u8,
+        keylen: size_t,
+        iv: *const u8,
+        ivlen: size_t,
+        out: *mut u8,
+    ) -> c_int;
+    fn get_aes_mac_gcm_len() -> c_uint;
+}
+
+pub fn aes_mac_gcm(data: &[u8], key: &[u8], iv: &[u8]) -> Result<Vec<u8>, i32> {
+    unsafe {
+        let mut ret = data.to_vec();
+        // let mut ret: Vec<u8> = Vec::with_capacity(get_aes_mac_gcm_len() as usize);
+        let out = ret.as_mut_ptr();
```

Comment:
For some reason the mut pointer of ret is NULL (within the c code), when it does not contain any values